src/Adapter/Product/CommandHandler/UpdateProductOptionsHandler.php

```diff
@@ -109,11 +109,11 @@ private function fillUpdatableProperties(Product $product, UpdateProductOptionsC
         }
         $availableForOrder = $product->available_for_order;
 
-        if (null !== $command->showPrice() && $availableForOrder) {
+        if (null !== $command->showPrice() && !$availableForOrder) {
             $product->show_price = $command->showPrice();
             $updatableProperties[] = 'show_price';
-        } elseif (!$availableForOrder && $product->show_price) {
-            $product->show_price = false;
+        } elseif ($availableForOrder && !$product->show_price) {
+            $product->show_price = true;
             $updatableProperties[] = 'show_price';
         }
```

Comment:
not sure if we need to allow changing the show_price when available for order is ON (cuz I think we must force show price ON too). I have done it this way (idk if its any better, gonna leave it hang here to consider :man_shrugging: )
```
        if ($availableForOrder) {
            // always force show price to true, when product is available for order
            $product->show_price = true;
            $updatableProperties[] = 'show_price';
        } elseif (null !== $command->showPrice()) {
            // if available for order is false, then we can update "showPrice" using value from command
            $product->show_price = $command->showPrice();
            $updatableProperties[] = 'show_price';
        }
```

Reply:
Actually both work, but the current implementation (which you did by the way ^^) has the advantage of only updating the field when it needs to so it's a bit more optimized